app/controllers/concerns/curation_concerns/single_use_links_controller_behavior.rb

```diff
@@ -40,11 +41,11 @@ def destroy
     protected
 
       def authorize_user!
-        authorize! :edit, asset
+        authorize! :edit, params[:id]
       end
 
-      def asset
-        @asset ||= ActiveFedora::Base.load_instance_from_solr(params[:id])
+      def asset_show_path
+        polymorphic_path([main_app, fetch(params[:id]).last])
```

Comment:
`#fetch()` returns a tuple, the first element is the solr response and the second is the solr document, which we want.